modules/moderation.py

```diff
@@ -148,6 +158,58 @@ async def find_discord_tokens(self, message: discord.Message) -> None:
         )
         await message.reply(msg)
 
+    async def pull_badbin_content(self, site: str, slug: str, *, fail_hard: bool = True) -> str:
+        async with self.bot.session.get(f"https://{site}/raw/{slug}") as f:
+            if 200 > f.status > 299:
+                if fail_hard:
+                    f.raise_for_status()
+                else:
+                    err = f"Could not read {slug} from {site}. Details: {f.status}\n\n{await f.read()}"
+                    logger.error(err)
+                    return err  # if we don't fail hard, we'll return the error message in the new paste.
+
+            return (await f.read()).decode()
+
+    async def post_mystbin_content(self, contents: list[tuple[str, str]]) -> tuple[str, str | None]:
+        hdrs = {}
+        if core.CONFIG["TOKENS"]["mystbin"]:
+            hdrs["Authorization"] = core.CONFIG["TOKENS"]["mystbin"]
+
+        async with self.bot.session.put(
+            "https://api.mystb.in/paste",
+            headers=hdrs,
+            json={"files": [{"filename": x[1], "content": x[0]} for x in contents]},
+        ) as resp:
+            if 200 > resp.status > 299:
+                resp.raise_for_status()
+
+            data = await resp.json()
+            return data["id"], data.get("notice") or None
+
+    @commands.Cog.listener("on_message")
+    async def find_badbins(self, message: discord.Message) -> None:
+        matches = self.BADBIN_RE.findall(message.content)
+
+        if matches:
+            contents: list[tuple[str, str]] = []
+
+            for match in matches:
+                site, slug, ext = match
+
+                if site is None or slug is None:
+                    continue
+
+                contents.append((await self.pull_badbin_content(site, slug), f"migrated.{ext or 'txt'}"))
+
+            if contents:
+                key, notice = await self.post_mystbin_content(contents)
```

Comment:
This whole thing can be replaced with `Bot.mb_client.create_paste` which already exists.